boefjes/boefjes/plugins/kat_dns_version/main.py

```diff
@@ -0,0 +1,49 @@
+"""Boefje script for getting namserver version"""
+
+import json
+import logging
+from os import getenv
+
+import dns
+
+from boefjes.job_models import BoefjeMeta
+
+logger = logging.getLogger(__name__)
+
+
+def run(boefje_meta: BoefjeMeta) -> list[tuple[set, str | bytes]]:
+    input_ = boefje_meta.arguments["input"]  # input is IPService
+    ip_port = input_["ip_port"]
+    if input_["service"]["name"] != "dns":
+        logging.info("Not a DNS service")
+        raise StopIteration
+
+    ip = ip_port["address"]["address"]
+    port = ip_port["port"]
+    protocol = ip_port["protocol"]
+
+    timeout = getenv("TIMEOUT", 30)
+
+    method = dns.query.tcp
+    if protocol == "udp":
+        method = dns.query.udp
+
+    queries = [
+        dns.message.make_query("VERSION.BIND", dns.rdatatype.TXT, dns.rdataclass.CHAOS),
+        dns.message.make_query("VERSION.SERVER", dns.rdatatype.TXT, dns.rdataclass.CHAOS),
+    ]
+
+    results = []
+    for query in queries:
+        response = method(query, where=ip, timeout=timeout, port=port)
+
+        try:
+            answer = response.answer[0]
+            results.append(answer.to_rdataset().pop().strings[0].decode())
+        except IndexError:
+            pass
+
+    if not results:
+        raise StopIteration
```

Comment:
It doesn't make sense to use `StopIteration` here, since this function is not a generator.

You might want to remove this if-clause and return an empty result; or use another error mechanism

Reply:
oops, wrong file, i was thinking i was in the normalizer generator.